diff --git a/src/lib.rs b/src/lib.rs
index 7e04986..482733a 100644
--- a/src/lib.rs
+++ b/src/lib.rs
@@ -61,11 +61,7 @@ pub mod pallet {
 	pub enum Event<T: Config> {
 		Created { owner: T::AccountId },
 		Transferred { from: T::AccountId, to: T::AccountId, kitty_id: [u8; 16] },
-		/* TODO: Create a new `Event` called `PriceSet` with fields:
-			- `owner` which is `T::AccountId`.
-			- `kitty_id` which is `[u8; 16]`.
-			- `new_price` which is `Option<BalanceOf<T>>`.
-		*/
+		PriceSet { owner: T::AccountId, kitty_id: [u8; 16], new_price: Option<BalanceOf<T>> },
 	}
 
 	#[pallet::error]
@@ -99,17 +95,15 @@ pub mod pallet {
 			Ok(())
 		}
 
-		/* TODO: Make an callable function called `set_price`:
-			- Inputs to the function are:
-				- `origin` which is `OriginFor<T>`.
-				- `kitty_id` which is `[u8; 16]`.
-				- `new_price` which is `Option<BalanceOf<T>`.
-			- Returns a `DispatchResult`
-			- The internal logic, for now, should be:
-				- Extract the caller `who` with `ensure_signed`.
-				- Call `Self::do_set_price` with the appropriate parameters, propagating the result.
-				- Return `Ok(())`.
-		*/
+		pub fn set_price(
+			origin: OriginFor<T>,
+			kitty_id: [u8; 16],
+			new_price: Option<BalanceOf<T>>,
+		) -> DispatchResult {
+			let who = ensure_signed(origin)?;
+			Self::do_set_price(who, kitty_id, new_price)?;
+			Ok(())
+		}
 	}
 
 	// Learn about internal functions.
@@ -175,15 +169,13 @@ pub mod pallet {
 			Ok(())
 		}
 
-		/* TODO: Make an internal function called `do_set_price`:
-			- Inputs to the function are:
-				- `caller` which is `T::AccountId`.
-				- `kitty_id` which is `[u8; 16]`.
-				- `new_price` which is `Option<BalanceOf<T>`.
-			- Returns a `DispatchResult`.
-			- The internal logic, for now, should be:
-				- `Self::deposit_event` for `Event::<T>::PriceSet` with the appropriate params.
-				- Return `Ok(())`.
-		*/
+		pub fn do_set_price(
+			caller: T::AccountId,
+			kitty_id: [u8; 16],
+			new_price: Option<BalanceOf<T>>,
+		) -> DispatchResult {
+			Self::deposit_event(Event::<T>::PriceSet { owner: caller, kitty_id, new_price });
+			Ok(())
+		}
 	}
 }
