diff --git a/src/lib.rs b/src/lib.rs
index 7429d13..44c72dc 100644
--- a/src/lib.rs
+++ b/src/lib.rs
@@ -32,6 +32,7 @@ pub mod pallet {
 	#[pallet::call]
 	impl<T: Config> Pallet<T> {
 		pub fn create_kitty(origin: OriginFor<T>) -> DispatchResult {
+			// Learn about `origin`.
 			let who = ensure_signed(origin)?;
 			Self::mint(who);
 			Ok(())
@@ -40,6 +41,7 @@ pub mod pallet {
 
 	// Learn about internal functions.
 	impl<T: Config> Pallet<T> {
+		// Learn about `AccountId`.
 		fn mint(owner: T::AccountId) {
 			Self::deposit_event(Event::<T>::Created { owner });
 		}
