diff --git a/src/lib.rs b/src/lib.rs
index e78b63b..3bf8fdb 100644
--- a/src/lib.rs
+++ b/src/lib.rs
@@ -32,6 +32,7 @@ pub mod pallet {
 	#[pallet::call]
 	impl<T: Config> Pallet<T> {
 		pub fn hello_world(origin: OriginFor<T>) -> DispatchResult {
+			// Learn about `origin`.
 			let who = ensure_signed(origin)?;
 			Self::say_hello(who);
 			Ok(())
@@ -40,6 +41,7 @@ pub mod pallet {
 
 	// Learn about internal functions.
 	impl<T: Config> Pallet<T> {
+		// Learn about `AccountId`.
 		fn say_hello(who: T::AccountId) {
 			Self::deposit_event(Event::<T>::Hello { who });
 		}
