diff --git a/src/impls.rs b/src/impls.rs
index fac2ead..ff69cae 100644
--- a/src/impls.rs
+++ b/src/impls.rs
@@ -1,18 +1,22 @@
 use super::*;
 use frame::prelude::*;
-/* 🚧 TODO 🚧: Import `frame::primitives::BlakeTwo256`. */
-/* 🚧 TODO 🚧: Import `frame::traits::Hash`. */
+use frame::primitives::BlakeTwo256;
+use frame::traits::Hash;
 
 impl<T: Config> Pallet<T> {
-	/* 🚧 TODO 🚧: Create a function `gen_dna` which returns a `[u8; 32]`.
-		- Create a `unique_payload` which contains data from `frame_system::Pallet::<T>`:
-			- `parent_hash`
-			- `block_number`
-			- `extrinsic_index`
-			- `CountForKitties::<T>::get()`
-		- Use `BlakeTwo256` to calculate the `hash_of` the unique payload.
-		- Return the hash as a `[u8; 32]`.
-	*/
+	// Generates and returns DNA
+	pub fn gen_dna() -> [u8; 32] {
+		// Create randomness payload. Multiple kitties can be generated in the same block,
+		// retaining uniqueness.
+		let unique_payload = (
+			frame_system::Pallet::<T>::parent_hash(),
+			frame_system::Pallet::<T>::block_number(),
+			frame_system::Pallet::<T>::extrinsic_index(),
+			CountForKitties::<T>::get(),
+		);
+
+		BlakeTwo256::hash_of(&unique_payload).into()
+	}
 
 	pub fn mint(owner: T::AccountId, dna: [u8; 32]) -> DispatchResult {
 		let kitty = Kitty { dna, owner: owner.clone() };
diff --git a/src/lib.rs b/src/lib.rs
index 34507f3..8795b0e 100644
--- a/src/lib.rs
+++ b/src/lib.rs
@@ -48,8 +48,7 @@ pub mod pallet {
 	impl<T: Config> Pallet<T> {
 		pub fn create_kitty(origin: OriginFor<T>) -> DispatchResult {
 			let who = ensure_signed(origin)?;
-			/* 🚧 TODO 🚧: Use the `Self::gen_dna()` function to generate a unique Kitty. */
-			let dna = [0u8; 32];
+			let dna = Self::gen_dna();
 			Self::mint(who, dna)?;
 			Ok(())
 		}
diff --git a/src/tests.rs b/src/tests.rs
index 53a248b..e2b802b 100644
--- a/src/tests.rs
+++ b/src/tests.rs
@@ -202,3 +202,15 @@ fn mint_stores_owner_in_kitty() {
 		assert_eq!(kitty.dna, [42u8; 32]);
 	})
 }
+
+#[test]
+fn create_kitty_makes_unique_kitties() {
+	new_test_ext().execute_with(|| {
+		// Two calls to `create_kitty` should work.
+		assert_ok!(PalletKitties::create_kitty(RuntimeOrigin::signed(ALICE)));
+		assert_ok!(PalletKitties::create_kitty(RuntimeOrigin::signed(BOB)));
+		// And should result in two kitties in our system.
+		assert_eq!(CountForKitties::<TestRuntime>::get(), 2);
+		assert_eq!(Kitties::<TestRuntime>::iter().count(), 2);
+	})
+}
