diff --git a/src/lib.rs b/src/lib.rs
index 5c8c3d6..6730289 100644
--- a/src/lib.rs
+++ b/src/lib.rs
@@ -40,8 +40,7 @@ pub mod pallet {
 	#[pallet::storage]
 	pub(super) type KittiesOwned<T: Config> = StorageMap<
 		Key = T::AccountId,
-		/* TODO: Turn this into a `BoundedVec` with a limit of `ConstU32<100>`. */
-		Value = Vec<[u8; 16]>,
+		Value = BoundedVec<[u8; 16], ConstU32<100>>,
 		QueryKind = ValueQuery,
 	>;
 
@@ -56,7 +55,7 @@ pub mod pallet {
 	pub enum Error<T> {
 		TooManyKitties,
 		DuplicateKitty,
-		/* TODO: Add a new `Error` named `TooManyOwned` */
+		TooManyOwned,
 	}
 
 	// Learn about callable functions and dispatch.
@@ -97,8 +96,7 @@ pub mod pallet {
 			let current_count: u64 = CountForKitties::<T>::get();
 			let new_count = current_count.checked_add(1).ok_or(Error::<T>::TooManyKitties)?;
 
-			/* TODO: Update `append` to `try_append` and `map_err` to `Error::<T>::TooManyOwned`. */
-			KittiesOwned::<T>::append(&owner, dna);
+			KittiesOwned::<T>::try_append(&owner, dna).map_err(|_| Error::<T>::TooManyOwned)?;
 			Kitties::<T>::insert(dna, kitty);
 			CountForKitties::<T>::set(new_count);
 
