diff --git a/src/impls.rs b/src/impls.rs
index 300924a..b2b0d67 100644
--- a/src/impls.rs
+++ b/src/impls.rs
@@ -34,4 +34,15 @@ impl<T: Config> Pallet<T> {
 		Self::deposit_event(Event::<T>::Created { owner });
 		Ok(())
 	}
+
+	/* TODO: Create an internal function called `do_transfer`:
+		- It has inputs:
+			- `from` which is `T::AccountId`.
+			- `to` which is `T::AccountId`.
+			- `kitty_id` which is `[u8; 16]`.
+		- It returns a `DispatchResult`
+		- The inner logic for now is:
+			- Call `Self::dispatch_event` on and emit `Event::<T>:Transferred` with params.
+			- Return `Ok(())`.
+	*/
 }
diff --git a/src/lib.rs b/src/lib.rs
index 8209cc2..2c51fa7 100644
--- a/src/lib.rs
+++ b/src/lib.rs
@@ -51,6 +51,12 @@ pub mod pallet {
 	#[pallet::generate_deposit(pub(super) fn deposit_event)]
 	pub enum Event<T: Config> {
 		Created { owner: T::AccountId },
+		/* TODO: Create a new event called `Transferred`:
+			- Parameters are:
+				- `from` which is `T::AccountId`.
+				- `to` which is `T::AccountId`.
+				- `kitty_id` which is `[u8; 16]`.
+		*/
 	}
 
 	#[pallet::error]
@@ -70,5 +76,17 @@ pub mod pallet {
 			Self::mint(who, dna)?;
 			Ok(())
 		}
+
+		/* TODO: Make a new extrinsic called `transfer`.
+			- Input parameters are:
+				- `origin` which is `OriginFor<T>`.
+				- `to` which is `T::AccountId`.
+				- `kitty_id` which is `[u8; 16]`.
+			- Returns a `DispatchResult`.
+			- The inner logic should be:
+				- Get the caller `who` from `ensure_signed`.
+				- Call `Self::do_transfer`, and propagate the result.
+				- End with Ok(()).
+		*/
 	}
 }
