diff --git a/src/lib.rs b/src/lib.rs
index 689d8ba..3920278 100644
--- a/src/lib.rs
+++ b/src/lib.rs
@@ -49,12 +49,9 @@ pub mod pallet {
 	impl<T: Config> Pallet<T> {
 		// Learn about `AccountId`.
 		fn mint(owner: T::AccountId) {
-			/* TODO:
-				- `get` the current count of kitties.
-				- `unwrap_or` set the count to `0`.
-				- increment the count by one.
-				- `set` the new count of kitties.
-			*/
+			let current_count = CountForKitties::<T>::get().unwrap_or(0);
+			let new_count = current_count + 1;
+			CountForKitties::<T>::set(Some(new_count));
 			Self::deposit_event(Event::<T>::Created { owner });
 		}
 	}
