diff --git a/src/impls.rs b/src/impls.rs
index 49906c6..869c781 100644
--- a/src/impls.rs
+++ b/src/impls.rs
@@ -1,8 +1,8 @@
 use super::*;
 use frame_support::pallet_prelude::*;
 
-// Learn about internal functions.
 impl<T: Config> Pallet<T> {
+	// Learn about `AccountId`.
 	pub fn mint(owner: T::AccountId) -> DispatchResult {
 		Self::deposit_event(Event::<T>::Created { owner });
 		Ok(())
diff --git a/src/lib.rs b/src/lib.rs
index 613e92b..a0865fc 100644
--- a/src/lib.rs
+++ b/src/lib.rs
@@ -34,6 +34,7 @@ pub mod pallet {
 	#[pallet::call]
 	impl<T: Config> Pallet<T> {
 		pub fn create_kitty(origin: OriginFor<T>) -> DispatchResult {
+			// Learn about `origin`.
 			let who = ensure_signed(origin)?;
 			Self::mint(who)?;
 			Ok(())
