diff --git a/src/impls.rs b/src/impls.rs
index 8d6e03a..4704ae6 100644
--- a/src/impls.rs
+++ b/src/impls.rs
@@ -1,8 +1,8 @@
 use super::*;
 use frame_support::pallet_prelude::*;
 
-/* 🚧 TODO 🚧:  Learn about internal functions. */
 impl<T: Config> Pallet<T> {
+	/* 🚧 TODO 🚧: Learn about `AccountId`. */
 	pub fn mint(owner: T::AccountId) -> DispatchResult {
 		Self::deposit_event(Event::<T>::Created { owner });
 		Ok(())
diff --git a/src/lib.rs b/src/lib.rs
index f73f837..aa5d372 100644
--- a/src/lib.rs
+++ b/src/lib.rs
@@ -27,10 +27,10 @@ pub mod pallet {
 	#[pallet::error]
 	pub enum Error<T> {}
 
-	/* 🚧 TODO 🚧: Learn about callable functions and dispatch. */
 	#[pallet::call]
 	impl<T: Config> Pallet<T> {
 		pub fn create_kitty(origin: OriginFor<T>) -> DispatchResult {
+			/* 🚧 TODO 🚧: Learn about origin. */
 			let who = ensure_signed(origin)?;
 			Self::mint(who)?;
 			Ok(())
