diff --git a/src/lib.rs b/src/lib.rs
index d44d77d..5a4e765 100644
--- a/src/lib.rs
+++ b/src/lib.rs
@@ -4,6 +4,8 @@ mod impls;
 mod tests;
 
 use frame::prelude::*;
+/* 🚧 TODO 🚧: Import `frame::traits::fungible::Inspect`. */
+/* 🚧 TODO 🚧: Import `frame::traits::fungible::Mutate`. */
 pub use pallet::*;
 
 #[frame::pallet(dev_mode)]
@@ -16,6 +18,13 @@ pub mod pallet {
 	#[pallet::config]
 	pub trait Config: frame_system::Config {
 		type RuntimeEvent: From<Event<Self>> + IsType<<Self as frame_system::Config>::RuntimeEvent>;
+
+		/* 🚧 TODO 🚧:
+			- Create a new associated type named `NativeBalance`.
+			- Require that `NativeBalance` implements the following traits:
+				- `Inspect` which is generic over `Self::AccountId`.
+				- `Mutate` which is also generic over `Self::AccountId`.
+		*/
 	}
 
 	#[derive(Encode, Decode, TypeInfo, MaxEncodedLen)]
diff --git a/src/tests.rs b/src/tests.rs
index bc81a1d..ee10774 100644
--- a/src/tests.rs
+++ b/src/tests.rs
@@ -61,6 +61,7 @@ impl pallet_balances::Config for TestRuntime {
 // will also need to update this configuration to represent that.
 impl pallet_kitties::Config for TestRuntime {
 	type RuntimeEvent = RuntimeEvent;
+	/* 🚧 TODO 🚧: Assign associated type `NativeBalance` to `PalletBalances`. */
 }
 
 // We need to run most of our tests using this function: `new_test_ext().execute_with(|| { ... });`
