diff --git a/src/lib.rs b/src/lib.rs
index 8a4f9d0..6dbe1b6 100644
--- a/src/lib.rs
+++ b/src/lib.rs
@@ -20,6 +20,10 @@ pub mod pallet {
 		type RuntimeEvent: From<Event<Self>> + IsType<<Self as frame_system::Config>::RuntimeEvent>;
 	}
 
+	/// Learn about storage value.
+	#[pallet::storage]
+	pub(super) type CountForKitties<T: Config> = StorageValue<Value = u64>;
+
 	// Learn about events.
 	#[pallet::event]
 	#[pallet::generate_deposit(pub(super) fn deposit_event)]
