diff --git a/src/impls.rs b/src/impls.rs
index 869c781..67fd2a0 100644
--- a/src/impls.rs
+++ b/src/impls.rs
@@ -4,6 +4,12 @@ use frame_support::pallet_prelude::*;
 impl<T: Config> Pallet<T> {
 	// Learn about `AccountId`.
 	pub fn mint(owner: T::AccountId) -> DispatchResult {
+		/* TODO:
+			- `get` the current count of kitties.
+			- `unwrap_or` set the count to `0`.
+			- increment the count by one.
+			- `set` the new count of kitties.
+		*/
 		Self::deposit_event(Event::<T>::Created { owner });
 		Ok(())
 	}
