diff --git a/src/lib.rs b/src/lib.rs
index ae8a09bc..7d88fffb 100644
--- a/src/lib.rs
+++ b/src/lib.rs
@@ -18,6 +18,12 @@ pub mod pallet {
 		type RuntimeEvent: From<Event<Self>> + IsType<<Self as frame_system::Config>::RuntimeEvent>;
 	}
 
+	/* 🚧 TODO 🚧:
+		- Create a new `StorageValue` named `CountForKitties`.
+			- `CountForKitties` should be generic over `<T: Config>`.
+			- Set `Value` to `u32` to store that type.
+	*/
+
 	#[pallet::event]
 	#[pallet::generate_deposit(pub(super) fn deposit_event)]
 	pub enum Event<T: Config> {
