diff --git a/src/lib.rs b/src/lib.rs
index d1dd4c6..177be5e 100644
--- a/src/lib.rs
+++ b/src/lib.rs
@@ -38,6 +38,7 @@ pub mod pallet {
 	#[pallet::error]
 	pub enum Error<T> {
 		TooManyKitties,
+		/* TODO: Create a new error `DuplicateKitty`. */
 	}
 
 	// Learn about callable functions and dispatch.
@@ -56,6 +57,10 @@ pub mod pallet {
 	impl<T: Config> Pallet<T> {
 		// Learn about `AccountId`.
 		fn mint(owner: T::AccountId, dna: [u8; 16]) -> DispatchResult {
+			/* TODO:
+				- `ensure!` that `Kitties` map does not `contains_key` for `dna`.
+				- If it does, return `Error::<T>::DuplicateKitty`.
+			*/
 			let current_count = CountForKitties::<T>::get().unwrap_or(0);
 			let new_count = current_count.checked_add(1).ok_or(Error::<T>::TooManyKitties)?;
 			Kitties::<T>::insert(dna, ());
