diff --git a/src/lib.rs b/src/lib.rs
index 75a1b65..c8ea870 100644
--- a/src/lib.rs
+++ b/src/lib.rs
@@ -18,6 +18,14 @@ pub mod pallet {
 		type RuntimeEvent: From<Event<Self>> + IsType<<Self as frame_system::Config>::RuntimeEvent>;
 	}
 
+	/* 🚧 TODO 🚧:
+		- Create a new `struct` called `Kitty`.
+		- Make `Kitty` generic over `T` where `T: Config`.
+		- Add two fields to `Kitty`:
+			- `dna` which is type `[u8; 32]`.
+			- `owner` which is type `T::AccountId`.
+	*/
+
 	#[pallet::storage]
 	pub(super) type CountForKitties<T: Config> = StorageValue<Value = u32, QueryKind = ValueQuery>;
 
diff --git a/src/tests.rs b/src/tests.rs
index ea6bb5b..de02e5c 100644
--- a/src/tests.rs
+++ b/src/tests.rs
@@ -28,6 +28,9 @@ type Block = frame_system::mocking::MockBlock<TestRuntime>;
 // We create the constants `ALICE` and `BOB` to make it clear when we are representing users below.
 const ALICE: u64 = 1;
 const BOB: u64 = 2;
+/* 🚧 TODO 🚧:
+	- Create `const DEFAULT_KITTY` with a `dna` of [0u8; 32] and an `owner` of 0 };.
+*/
 
 #[runtime]
 mod runtime {
