diff --git a/src/impls.rs b/src/impls.rs
index 5ec04ae..686cb0a 100644
--- a/src/impls.rs
+++ b/src/impls.rs
@@ -57,4 +57,15 @@ impl<T: Config> Pallet<T> {
 		Self::deposit_event(Event::<T>::Transferred { from, to, kitty_id });
 		Ok(())
 	}
+
+	/* 🚧 TODO 🚧: Make an internal function called `do_set_price`:
+		- Inputs to the function are:
+			- `caller` which is `T::AccountId`.
+			- `kitty_id` which is `[u8; 32]`.
+			- `new_price` which is `Option<BalanceOf<T>`.
+		- Returns a `DispatchResult`.
+		- The internal logic, for now, should be:
+			- `Self::deposit_event` for `Event::<T>::PriceSet` with the appropriate params.
+			- Return `Ok(())`.
+	*/
 }
diff --git a/src/lib.rs b/src/lib.rs
index f2625a6..53bb1ec 100644
--- a/src/lib.rs
+++ b/src/lib.rs
@@ -54,6 +54,11 @@ pub mod pallet {
 	pub enum Event<T: Config> {
 		Created { owner: T::AccountId },
 		Transferred { from: T::AccountId, to: T::AccountId, kitty_id: [u8; 32] },
+		/* 🚧 TODO 🚧: Create a new `Event` called `PriceSet` with fields:
+			- `owner` which is `T::AccountId`.
+			- `kitty_id` which is `[u8; 32]`.
+			- `new_price` which is `Option<BalanceOf<T>>`.
+		*/
 	}
 
 	#[pallet::error]
@@ -84,5 +89,17 @@ pub mod pallet {
 			Self::do_transfer(who, to, kitty_id)?;
 			Ok(())
 		}
+
+		/* 🚧 TODO 🚧: Make an callable function called `set_price`:
+			- Inputs to the function are:
+				- `origin` which is `OriginFor<T>`.
+				- `kitty_id` which is `[u8; 32]`.
+				- `new_price` which is `Option<BalanceOf<T>`.
+			- Returns a `DispatchResult`
+			- The internal logic, for now, should be:
+				- Extract the caller `who` with `ensure_signed`.
+				- Call `Self::do_set_price` with the appropriate parameters, propagating the result.
+				- Return `Ok(())`.
+		*/
 	}
 }
