diff --git a/src/impls.rs b/src/impls.rs
index c990886..a8ea817 100644
--- a/src/impls.rs
+++ b/src/impls.rs
@@ -3,11 +3,16 @@ use frame::prelude::*;
 
 impl<T: Config> Pallet<T> {
 	pub fn mint(owner: T::AccountId, dna: [u8; 32]) -> DispatchResult {
+		/* 🚧 TODO 🚧:
+			- Create a new variable `kitty` which is a `Kitty` struct with `dna` and `owner`.
+		*/
+
 		// Check if the kitty does not already exist in our storage map
 		ensure!(!Kitties::<T>::contains_key(dna), Error::<T>::DuplicateKitty);
 
 		let current_count: u64 = CountForKitties::<T>::get();
 		let new_count = current_count.checked_add(1).ok_or(Error::<T>::TooManyKitties)?;
+		/* 🚧 TODO 🚧: Insert `kitty`into the map instead of `()`. */
 		Kitties::<T>::insert(dna, ());
 		CountForKitties::<T>::set(new_count);
 		Self::deposit_event(Event::<T>::Created { owner });
diff --git a/src/lib.rs b/src/lib.rs
index 8349e90..59cc20b 100644
--- a/src/lib.rs
+++ b/src/lib.rs
@@ -17,6 +17,10 @@ pub mod pallet {
 		type RuntimeEvent: From<Event<Self>> + IsType<<Self as frame_system::Config>::RuntimeEvent>;
 	}
 
+	/* 🚧 TODO 🚧:
+		- Add the derive macros needed for putting a struct in storage.
+		- Add `#[scale_info(skip_type_params(T))]` to ignore the generic `T`.
+	*/
 	pub struct Kitty<T: Config> {
 		// Using 32 bytes to represent a kitty DNA
 		pub dna: [u8; 32],
@@ -27,6 +31,7 @@ pub mod pallet {
 	pub(super) type CountForKitties<T: Config> = StorageValue<Value = u64, QueryKind = ValueQuery>;
 
 	#[pallet::storage]
+	/* 🚧 TODO 🚧: Update the `Value` to be type `Kitty<T>` instead of (). */
 	pub(super) type Kitties<T: Config> = StorageMap<Key = [u8; 32], Value = ()>;
 
 	#[pallet::event]
