diff --git a/src/lib.rs b/src/lib.rs
index 7142e28..5c8c3d6 100644
--- a/src/lib.rs
+++ b/src/lib.rs
@@ -38,8 +38,12 @@ pub mod pallet {
 
 	/// Track the kitties owned by each account.
 	#[pallet::storage]
-	pub(super) type KittiesOwned<T: Config> =
-		StorageMap<Key = T::AccountId, Value = Vec<[u8; 16]>, QueryKind = ValueQuery>;
+	pub(super) type KittiesOwned<T: Config> = StorageMap<
+		Key = T::AccountId,
+		/* TODO: Turn this into a `BoundedVec` with a limit of `ConstU32<100>`. */
+		Value = Vec<[u8; 16]>,
+		QueryKind = ValueQuery,
+	>;
 
 	// Learn about events.
 	#[pallet::event]
@@ -52,6 +56,7 @@ pub mod pallet {
 	pub enum Error<T> {
 		TooManyKitties,
 		DuplicateKitty,
+		/* TODO: Add a new `Error` named `TooManyOwned` */
 	}
 
 	// Learn about callable functions and dispatch.
@@ -92,6 +97,7 @@ pub mod pallet {
 			let current_count: u64 = CountForKitties::<T>::get();
 			let new_count = current_count.checked_add(1).ok_or(Error::<T>::TooManyKitties)?;
 
+			/* TODO: Update `append` to `try_append` and `map_err` to `Error::<T>::TooManyOwned`. */
 			KittiesOwned::<T>::append(&owner, dna);
 			Kitties::<T>::insert(dna, kitty);
 			CountForKitties::<T>::set(new_count);
