diff --git a/src/lib.rs b/src/lib.rs
index 59a65fe..924f26f 100644
--- a/src/lib.rs
+++ b/src/lib.rs
@@ -32,7 +32,11 @@ pub mod pallet {
 	}
 
 	#[pallet::error]
-	pub enum Error<T> {}
+	pub enum Error<T> {
+		/* TODO:
+			- Introduce a new error `TooManyKitties`.
+		*/
+	}
 
 	// Learn about callable functions and dispatch.
 	#[pallet::call]
@@ -50,6 +54,7 @@ pub mod pallet {
 		// Learn about `AccountId`.
 		fn mint(owner: T::AccountId) -> DispatchResult {
 			let current_count: u64 = CountForKitties::<T>::get().unwrap_or(0);
+			/* TODO: Update this logic to use safe math. */
 			let new_count = current_count + 1;
 			CountForKitties::<T>::set(Some(new_count));
 			Self::deposit_event(Event::<T>::Created { owner });
