diff --git a/src/lib.rs b/src/lib.rs
index 022033d..4b7e602 100644
--- a/src/lib.rs
+++ b/src/lib.rs
@@ -6,7 +6,10 @@ pub use pallet::*;
 #[frame_support::pallet(dev_mode)]
 pub mod pallet {
 	use super::*;
-	use frame_support::pallet_prelude::*;
+	use frame_support::{
+		pallet_prelude::*,
+		traits::fungible::{Inspect, Mutate},
+	};
 	use frame_system::pallet_prelude::*;
 
 	// Learn about the Pallet struct: the structure on which we implement all functions and traits
@@ -18,6 +21,9 @@ pub mod pallet {
 	#[pallet::config]
 	pub trait Config: frame_system::Config {
 		type RuntimeEvent: From<Event<Self>> + IsType<<Self as frame_system::Config>::RuntimeEvent>;
+
+		/// The Fungible handler for the kitties pallet.
+		type NativeBalance: Inspect<Self::AccountId> + Mutate<Self::AccountId>;
 	}
 
 	#[derive(Encode, Decode, Clone, Copy, PartialEq, RuntimeDebug, TypeInfo, MaxEncodedLen)]
