diff --git a/src/lib.rs b/src/lib.rs
index 67dd167..356e547 100644
--- a/src/lib.rs
+++ b/src/lib.rs
@@ -49,6 +49,12 @@ pub mod pallet {
 	impl<T: Config> Pallet<T> {
 		// Learn about `AccountId`.
 		fn mint(owner: T::AccountId) -> DispatchResult {
+			/* TODO:
+				- `get` the current count of kitties.
+				- `unwrap_or` set the count to `0`.
+				- increment the count by one.
+				- `set` the new count of kitties.
+			*/
 			Self::deposit_event(Event::<T>::Created { owner });
 			Ok(())
 		}
