diff --git a/src/impls.rs b/src/impls.rs
index d73e0e1..a68886b 100644
--- a/src/impls.rs
+++ b/src/impls.rs
@@ -71,14 +71,12 @@ impl<T: Config> Pallet<T> {
 		Ok(())
 	}
 
-	/* 🚧 TODO 🚧: Create a new internal function `do_buy_kitty`:
-		- Inputs to the function are:
-			- `buyer` which is `T::AccountId`.
-			- `kitty_id` which is `[u8; 16]`.
-			- `price` which is `BalanceOf<T>`.
-		- It returns `DispatchResult`.
-		- The internal logic, for now, should be:
-			- `Self::deposit_event` with `Event::<T>::Sold` and the appropriate params.
-			- Return `Ok(())`.
-	*/
+	pub fn do_buy_kitty(
+		buyer: T::AccountId,
+		kitty_id: [u8; 16],
+		price: BalanceOf<T>,
+	) -> DispatchResult {
+		Self::deposit_event(Event::<T>::Sold { buyer, kitty_id, price });
+		Ok(())
+	}
 }
diff --git a/src/lib.rs b/src/lib.rs
index 20a1172..dde08bf 100644
--- a/src/lib.rs
+++ b/src/lib.rs
@@ -57,11 +57,7 @@ pub mod pallet {
 		Created { owner: T::AccountId },
 		Transferred { from: T::AccountId, to: T::AccountId, kitty_id: [u8; 16] },
 		PriceSet { owner: T::AccountId, kitty_id: [u8; 16], new_price: Option<BalanceOf<T>> },
-		/* 🚧 TODO 🚧: Create a new `Event` called `Sold` with the following parameters:
-			- `buyer` which is `T::AccountId`.
-			- `kitty_id` which is `[u8; 16]`.
-			- `price` which is `BalanceOf<T>`.
-		*/
+		Sold { buyer: T::AccountId, kitty_id: [u8; 16], price: BalanceOf<T> },
 	}
 
 	#[pallet::error]
@@ -103,16 +99,14 @@ pub mod pallet {
 			Ok(())
 		}
 
-		/* 🚧 TODO 🚧: Create a new callable function `buy_kitty`:
-			- Inputs to the function are:
-				- `origin` which is `OriginFor<T>`.
-				- `kitty_id` which is `[u8; 16]`.
-				- `max_price` which is `BalanceOf<T>`.
-			- It returns `DispatchResult`.
-			- The internal logic should be:
-				- Extract `who` using `ensure_signed` on `origin`.
-				- Call `Self::do_buy_kitty` using appropriate params, and propagating the result.
-				- Return `Ok(())`.
-		*/
+		pub fn buy_kitty(
+			origin: OriginFor<T>,
+			kitty_id: [u8; 16],
+			max_price: BalanceOf<T>,
+		) -> DispatchResult {
+			let who = ensure_signed(origin)?;
+			Self::do_buy_kitty(who, kitty_id, max_price)?;
+			Ok(())
+		}
 	}
 }
