diff --git a/src/lib.rs b/src/lib.rs
index 5a4e7650..59150bd6 100644
--- a/src/lib.rs
+++ b/src/lib.rs
@@ -4,8 +4,8 @@ mod impls;
 mod tests;
 
 use frame::prelude::*;
-/* 🚧 TODO 🚧: Import `frame::traits::fungible::Inspect`. */
-/* 🚧 TODO 🚧: Import `frame::traits::fungible::Mutate`. */
+use frame::traits::fungible::Inspect;
+use frame::traits::fungible::Mutate;
 pub use pallet::*;
 
 #[frame::pallet(dev_mode)]
@@ -19,12 +19,8 @@ pub mod pallet {
 	pub trait Config: frame_system::Config {
 		type RuntimeEvent: From<Event<Self>> + IsType<<Self as frame_system::Config>::RuntimeEvent>;
 
-		/* 🚧 TODO 🚧:
-			- Create a new associated type named `NativeBalance`.
-			- Require that `NativeBalance` implements the following traits:
-				- `Inspect` which is generic over `Self::AccountId`.
-				- `Mutate` which is also generic over `Self::AccountId`.
-		*/
+		/// The Fungible handler for the kitties pallet.
+		type NativeBalance: Inspect<Self::AccountId> + Mutate<Self::AccountId>;
 	}
 
 	#[derive(Encode, Decode, TypeInfo, MaxEncodedLen)]
diff --git a/src/tests.rs b/src/tests.rs
index 8a76f591..df5dc0dd 100644
--- a/src/tests.rs
+++ b/src/tests.rs
@@ -81,7 +81,7 @@ impl pallet_balances::Config for TestRuntime {
 // will also need to update this configuration to represent that.
 impl pallet_kitties::Config for TestRuntime {
 	type RuntimeEvent = RuntimeEvent;
-	/* 🚧 TODO 🚧: Assign associated type `NativeBalance` to `PalletBalances`. */
+	type NativeBalance = PalletBalances;
 }
 
 // We need to run most of our tests using this function: `new_test_ext().execute_with(|| { ... });`
@@ -311,3 +311,14 @@ fn transfer_logic_works() {
 		assert_eq!(kitty.owner, BOB);
 	});
 }
+
+#[test]
+fn native_balance_associated_type_works() {
+	new_test_ext().execute_with(|| {
+		assert_ok!(<<TestRuntime as Config>::NativeBalance as Mutate<_>>::mint_into(&ALICE, 1337));
+		assert_eq!(
+			<<TestRuntime as Config>::NativeBalance as Inspect<_>>::total_balance(&ALICE),
+			1337
+		);
+	});
+}
