diff --git a/src/lib.rs b/src/lib.rs
index a52896ed..ae8a09bc 100644
--- a/src/lib.rs
+++ b/src/lib.rs
@@ -18,7 +18,6 @@ pub mod pallet {
 		type RuntimeEvent: From<Event<Self>> + IsType<<Self as frame_system::Config>::RuntimeEvent>;
 	}
 
-	/* 🚧 TODO 🚧: Learn about Pallet Events. */
 	#[pallet::event]
 	#[pallet::generate_deposit(pub(super) fn deposit_event)]
 	pub enum Event<T: Config> {
diff --git a/src/tests.rs b/src/tests.rs
index 434ce723..59889339 100644
--- a/src/tests.rs
+++ b/src/tests.rs
@@ -29,7 +29,6 @@ type Block = frame_system::mocking::MockBlock<TestRuntime>;
 const ALICE: u64 = 1;
 const BOB: u64 = 2;
 
-/* 🚧 TODO 🚧: Learn about constructing a runtime. */
 #[runtime]
 mod runtime {
 	#[runtime::derive(
@@ -61,7 +60,6 @@ mod runtime {
 	pub type PalletKitties = pallet_kitties::Pallet<TestRuntime>;
 }
 
-/* 🚧 TODO 🚧: Learn about configuring a pallet. */
 // Normally `System` would have many more configurations, but you can see that we use some macro
 // magic to automatically configure most of the pallet for a "default test configuration".
 #[derive_impl(frame_system::config_preludes::TestDefaultConfig)]
@@ -84,7 +82,6 @@ impl pallet_kitties::Config for TestRuntime {
 	type RuntimeEvent = RuntimeEvent;
 }
 
-/* 🚧 TODO 🚧: Learn about test externalities. */
 // We need to run most of our tests using this function: `new_test_ext().execute_with(|| { ... });`
 // It simulates the blockchain database backend for our tests.
 // If you forget to include this and try to access your Pallet storage, you will get an error like:
@@ -127,7 +124,6 @@ fn create_kitty_checks_signed() {
 	})
 }
 
-/* 🚧 TODO 🚧: Learn about writing tests. */
 #[test]
 fn create_kitty_emits_event() {
 	new_test_ext().execute_with(|| {
