diff --git a/src/lib.rs b/src/lib.rs
index 95953bf..91b5ca8 100644
--- a/src/lib.rs
+++ b/src/lib.rs
@@ -24,6 +24,10 @@ pub mod pallet {
 	#[pallet::storage]
 	pub(super) type CountForKitties<T: Config> = StorageValue<Value = u64>;
 
+	/// Learn about storage maps.
+	#[pallet::storage]
+	pub(super) type Kitties<T: Config> = StorageMap<Key = [u8; 16], Value = ()>;
+
 	// Learn about events.
 	#[pallet::event]
 	#[pallet::generate_deposit(pub(super) fn deposit_event)]
