diff --git a/src/impls.rs b/src/impls.rs
index 292909e..c0e7cb2 100644
--- a/src/impls.rs
+++ b/src/impls.rs
@@ -3,10 +3,9 @@ use frame::prelude::*;
 
 impl<T: Config> Pallet<T> {
 	pub fn mint(owner: T::AccountId, dna: [u8; 32]) -> DispatchResult {
-		/* 🚧 TODO 🚧:
-			- `ensure!` that `Kitties` map does not `contains_key` for `dna`.
-			- If it does, return `Error::<T>::DuplicateKitty`.
-		*/
+		// Check if the kitty does not already exist in our storage map
+		ensure!(!Kitties::<T>::contains_key(dna), Error::<T>::DuplicateKitty);
+
 		let current_count: u32 = CountForKitties::<T>::get();
 		let new_count = current_count.checked_add(1).ok_or(Error::<T>::TooManyKitties)?;
 		Kitties::<T>::insert(dna, ());
diff --git a/src/lib.rs b/src/lib.rs
index 6fb4b6d..75a1b65 100644
--- a/src/lib.rs
+++ b/src/lib.rs
@@ -33,7 +33,7 @@ pub mod pallet {
 	#[pallet::error]
 	pub enum Error<T> {
 		TooManyKitties,
-		/* 🚧 TODO 🚧: Create a new error `DuplicateKitty`. */
+		DuplicateKitty,
 	}
 
 	#[pallet::call]
diff --git a/src/tests.rs b/src/tests.rs
index 1da91ca..5caa134 100644
--- a/src/tests.rs
+++ b/src/tests.rs
@@ -172,3 +172,11 @@ fn create_kitty_adds_to_map() {
 		assert_eq!(Kitties::<TestRuntime>::iter().count(), 1);
 	})
 }
+
+#[test]
+fn cannot_mint_duplicate_kitty() {
+	new_test_ext().execute_with(|| {
+		assert_ok!(PalletKitties::mint(ALICE, [0u8; 32]));
+		assert_noop!(PalletKitties::mint(BOB, [0u8; 32]), Error::<TestRuntime>::DuplicateKitty);
+	})
+}
