diff --git a/src/lib.rs b/src/lib.rs
index a142105..5f5f631 100644
--- a/src/lib.rs
+++ b/src/lib.rs
@@ -22,13 +22,11 @@ pub mod pallet {
 		type RuntimeEvent: From<Event<Self>> + IsType<<Self as frame_system::Config>::RuntimeEvent>;
 	}
 
-	/* TODO:
-		- Create a new `struct` called `Kitty`.
-		- Make `Kitty` generic over `T` where `T: Config`.
-		- Add two fields to `Kitty`:
-			- `dna` which is type `[u8; 16]`.
-			- `owner` which is type `T::AccountId`.
-	*/
+	pub struct Kitty<T: Config> {
+		// Using 16 bytes to represent a kitty DNA
+		pub dna: [u8; 16],
+		pub owner: T::AccountId,
+	}
 
 	/// Learn about storage value.
 	#[pallet::storage]
