diff --git a/src/lib.rs b/src/lib.rs
index 52b2dba..5b597ae 100644
--- a/src/lib.rs
+++ b/src/lib.rs
@@ -20,6 +20,10 @@ pub mod pallet {
 	#[pallet::storage]
 	pub(super) type CountForKitties<T: Config> = StorageValue<Value = u32, QueryKind = ValueQuery>;
 
+	/* 🚧 TODO 🚧: Learn about storage maps. */
+	#[pallet::storage]
+	pub(super) type Kitties<T: Config> = StorageMap<Key = [u8; 32], Value = ()>;
+
 	#[pallet::event]
 	#[pallet::generate_deposit(pub(super) fn deposit_event)]
 	pub enum Event<T: Config> {
