diff --git a/src/lib.rs b/src/lib.rs
index bcfe4da..ab7c5ed 100644
--- a/src/lib.rs
+++ b/src/lib.rs
@@ -7,7 +7,10 @@ pub use pallet::*;
 #[frame_support::pallet(dev_mode)]
 pub mod pallet {
 	use super::*;
-	use frame_support::pallet_prelude::*;
+	use frame_support::{
+		pallet_prelude::*,
+		traits::fungible::{Inspect, Mutate},
+	};
 	use frame_system::pallet_prelude::*;
 
 	#[pallet::pallet]
@@ -16,6 +19,9 @@ pub mod pallet {
 	#[pallet::config]
 	pub trait Config: frame_system::Config {
 		type RuntimeEvent: From<Event<Self>> + IsType<<Self as frame_system::Config>::RuntimeEvent>;
+
+		/// The Fungible handler for the kitties pallet.
+		type NativeBalance: Inspect<Self::AccountId> + Mutate<Self::AccountId>;
 	}
 
 	#[derive(Encode, Decode, TypeInfo, MaxEncodedLen)]
