diff --git a/src/lib.rs b/src/lib.rs
index be05ddb..d4e9d01 100644
--- a/src/lib.rs
+++ b/src/lib.rs
@@ -20,12 +20,8 @@ pub mod pallet {
 	#[pallet::storage]
 	pub(super) type CountForKitties<T: Config> = StorageValue<Value = u32, QueryKind = ValueQuery>;
 
-	/* 🚧 TODO 🚧:
-		- Create a new `StorageMap` named `Kitties`.
-			- `Kitties` should be generic over `<T: Config>`.
-			- Set `Key` to `[u8; 32]` to use the kitty id as the key.
-			- Set `Value` to `()` as a placeholder for now.
-	*/
+	#[pallet::storage]
+	pub(super) type Kitties<T: Config> = StorageMap<Key = [u8; 32], Value = ()>;
 
 	#[pallet::event]
 	#[pallet::generate_deposit(pub(super) fn deposit_event)]
